### PR TITLE
Topic routing

### PR DESCRIPTION
**Pull request by annieblomgren**
### WHY are these changes introduced?

Other examples have routing key 'stats', so messages are not routed to the mqtt.subscription* queues. 

### WHAT is this pull request doing?

Adding an example called topic.js where the routing key is the topic so that messages are routed to the mqtt.subscription* queues.
Have added **qos** and **clean session** as parameters instead of fixed in the examples. 
QoS takes value 0 or 1 (value 2 will be downgraded to 1 by RabbitMQ).
Clean session takes value true or false. 

"Usage: node topic.js URI CONNECTIONS QOS (0 or 1) CLEAN (true or false) [PREFIX (defaults to HOSTNAME)]"

### HOW can this pull request be tested?

node topic.js $M 10 1 true

### CODE REVIEW

**Comment by dentarg** [comment]
qos 0 or 1 and clean true/false can be arguments (or ENV vars) so we don't have to duplicate code :)

**Comment by annieblomgren** [comment]
I have added qos and clean session as parameters now in file topic.js, does that look better @dentarg ? 
I will delete the separate qos files if it's better.

**Comment by dentarg** [comment]
Yes 👍

File: topic.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+if (process.argv.length < 6) {
+  console.log("Usage: node app.js URI CONNECTIONS QOS (0 or 1) CLEAN (true or false) [PREFIX (defaults to HOSTNAME)]")
+  process.exit()
+}
+const uri = process.argv[2]
+const conns = parseInt(process.argv[3])
+const prefix = process.argv[6] || os.hostname()
+const qos = parseInt(process.argv[4])
+const clean = parseInt(process.argv[5])
```
Comment (dentarg) [inline topic.js]:
```suggestion
const uri = process.argv[2]
const conns = parseInt(process.argv[3])
const qos = parseInt(process.argv[4])
const clean = parseInt(process.argv[5])
const prefix = process.argv[6] || os.hostname()
```

**Review by dentarg — APPROVED** [review]
I trust you have run this and are happy with it :)

We can always improve more later on if we find something

**Comment by dentarg** [comment]
Don't forget to update the pull request description now that we only have one file.

Then you can squash merge this (using the button here on GitHub)

**Pull request merged**